cirq-web/cirq_ts/src/bloch_sphere/bloch_sphere_test.ts

```diff
@@ -0,0 +1,245 @@
+      expect(horizontalMeridians.numCircles).to.equal(7);
+    });
+
+    it('contains 4 vertical meridians by default', () => {
+      expect(verticalMeridians.numCircles).to.equal(4);
+    });
+
+    it('horizontal chord meridians are gray by default', () => {
+      expect(horizontalMeridians.color).to.equal('gray');
+    });
+
+    it('vertical meridians are gray by default', () => {
+      expect(verticalMeridians.color).to.equal('gray');
+    });
+
+    it('horizontal chord meridians share the same radius (5) of the sphere by default', () => {
+      expect(horizontalMeridians.radius).to.equal(5);
+    });
+
+    it('vertical meridians share the same radius (5) of the sphere by default', () => {
+      expect(verticalMeridians.radius).to.equal(5);
+    });
+  });
+
+  describe('Axes', () => {
+    const children = bloch_sphere.children;
+    const axes = children.find(
+      child => child.constructor.name === 'Axes'
+    ) as Axes;
+
+    it('contains the 3 axes by default', () => {
+      const numberOfAxisLines = axes.children.length;
+      expect(numberOfAxisLines).to.equal(3);
+    });
+
+    it('The axes fit a circle of radius 5 by default', () => {
+      expect(axes.radius).to.equal(5);
+    });
+
+    it('The axes are the right colors by default', () => {
+      expect(axes.xAxisColor).to.equal('#1f51ff');
+      expect(axes.yAxisColor).to.equal('#ff3131');
+      expect(axes.zAxisColor).to.equal('#39ff14');
+    });
+  });
+
+  describe('Labels', () => {
+    const children = bloch_sphere.children;
+    const labels = children.find(
+      child => child.constructor.name === 'Labels'
+    ) as Labels;
+
+    it('The default labels are fit towards the default radius of the sphere (5) with the default spacing (0.5)', () => {
+      const expectedBlochSphereLabels = {
+        '|+⟩': new Vector3(5.5, 0, 0),
+        '|-⟩': new Vector3(-5.5, 0, 0),
+        '|i⟩': new Vector3(0, 0, -5.5),
+        '|-i⟩': new Vector3(0, 0, 5.5),
+        '|0⟩': new Vector3(0, 5.5, 0),
+        '|1⟩': new Vector3(0, -5.5, 0),
+      };
+      expect(labels.labels).to.eql(expectedBlochSphereLabels);
+    });
+  });
+});
+
+describe('Blochsphere configurables', () => {
+  const bloch_sphere = new BlochSphere(3, 9, 6);
+  const children = bloch_sphere.children;
+
+  const sphere = children.find(
+    child => child.constructor.name === 'Sphere'
+  ) as Sphere;
+
+  const meridians = children.filter(
+    child => child.constructor.name === 'Meridians'
+  ) as Meridians[];
+
+  const horizontalMeridians = meridians.find(
+    child => child.orientation === Orientation.HORIZONTAL_CHORD
+  ) as Meridians;
+
+  const verticalMeridians = meridians.find(
+    child => child.orientation === Orientation.VERTICAL
+  ) as Meridians;
+
+  describe('the radius is configurable', () => {
+    it('accepts a radius of 3', () => {
+      expect(sphere.radius).to.equal(3);
+    });
+
+    it('defaults correctly to an invalid radius', () => {
```

Comment:
"fail fast", throw an error instead of silently creating a seemingly valid object.